streams/test-utils/src/main/java/org/apache/kafka/streams/test/ConsumerRecordFactory.java

```diff
@@ -0,0 +1,414 @@
+        this(defaultTopicName, keySerializer, valueSerializer, startTimestampMs, 0L);
+    }
+
+    /**
+     * Create a new factory for the given topic.
+     *
+     * @param keySerializer the key serializer
+     * @param valueSerializer the value serializer
+     * @param startTimestampMs the initial timestamp for generated records
+     * @param autoAdvanceMs the time increment pre generated record
+     */
+    public ConsumerRecordFactory(final Serializer<K> keySerializer,
+                                 final Serializer<V> valueSerializer,
+                                 final long startTimestampMs,
+                                 final long autoAdvanceMs) {
+        this(null, keySerializer, valueSerializer, startTimestampMs, autoAdvanceMs);
+    }
+
+    /**
+     * Create a new factory for the given topic.
+     *
+     * @param defaultTopicName the topic name used for all generated {@link ConsumerRecord consumer records}
+     * @param keySerializer the key serializer
+     * @param valueSerializer the value serializer
+     * @param startTimestampMs the initial timestamp for generated records
+     * @param autoAdvanceMs the time increment pre generated record
+     */
+    public ConsumerRecordFactory(final String defaultTopicName,
+                                 final Serializer<K> keySerializer,
+                                 final Serializer<V> valueSerializer,
+                                 final long startTimestampMs,
+                                 final long autoAdvanceMs) {
+        Objects.requireNonNull(keySerializer, "keySerializer cannot be null");
+        Objects.requireNonNull(valueSerializer, "valueSerializer cannot be null");
+        this.topicName = defaultTopicName;
+        this.keySerializer = keySerializer;
+        this.valueSerializer = valueSerializer;
+        timeMs = startTimestampMs;
+        advanceMs = autoAdvanceMs;
+    }
+
+    /**
+     * Advances the internally tracked time.
+     *
+     * @param advanceMs the amount of time to advance
+     */
+    public void advanceTimeMs(final long advanceMs) {
+        if (advanceMs < 0) {
+            throw new IllegalArgumentException("advanceMs must be positive");
+        }
+        timeMs += advanceMs;
+    }
+
+    /**
+     * Create a {@link ConsumerRecord} with the given topic name, key, value, and timestamp.
+     * Does not auto advance internally tracked time.
+     *
+     * @param topicName the topic name
+     * @param key the record key
+     * @param value the record value
+     * @param timestampMs the record timestamp
+     * @return the generated {@link ConsumerRecord}
+     */
+    public ConsumerRecord<byte[], byte[]> create(final String topicName,
+                                                 final K key,
+                                                 final V value,
+                                                 final long timestampMs) {
+        if (topicName == null) {
```

Comment:
`Objects.requireNonNull(..)` ?

Reply:
We could. No strong opinion -- `IllegalArgumentException` seems more appropriate than NPE.

Reply:
It would be consistent with what we have done elsewhere in the API